test/hotspot/jtreg/runtime/linkResolver/NoSuchFieldOutputTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8298065
+ * @summary Test output of NoSuchFieldError when field signature does not match
+ * @compile NoSuchFieldOutputTest.java FieldName.java
+ * @compile FieldName1.jasm
+ * @run main NoSuchFieldOutputTest
+ * @compile FieldName2.jasm
+ * @run main NoSuchFieldOutputTest
+ * @compile FieldName3.jasm
+ * @run main NoSuchFieldOutputTest
+ */
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class NoSuchFieldOutputTest {
+
+  public static void main(java.lang.String[] unused) throws Exception {
+    try {
+      FieldName fm = new FieldName();
+      String s = "";
+      Object x = FieldName.x;
+      s = s + "\nx = " + x;
+      Object y = FieldName.y;
+      s = s + "\ny = " + y;
+      Object z = FieldName.z;
+      s = s + "\nz = " + z;
+      throwTestException("Did not throw NoSuchFieldError", s);
+    } catch (NoSuchFieldError nsfe) {
```

Comment:
You could put each field access in a try block and remove all the fields in the second jasm file, so you can test for all the messages specifically.  Although the pattern compiler is cool.

Reply:
Compiling each jasm file after the main file, IS the easiest way to run it outside jtreg.

Reply:
Here we have very simple conditions to test, so there's no need for NoSuchFieldOutputTest.java to access FieldName.java at all. That's why I propose that all the error conditions should be implemented inside the jasm files alone. NoSuchFieldOutputTest.java can just refer to the error class by name.

Reply:
Okay so rather than generate the NSFE by separate compilation, Ioi is suggesting the jasm code can reference a non-existent field directly and so throw NSFE. That works too and the test can be re-run stand-alone.